<?xml version="1.0" encoding="UTF-8"?>
<!-- 
    This is the production launch file to be used with the real roboclaw device.
-->
 <launch>

    <arg name="name"            default="roboclaw"                  />
    <arg name="test_mode"       default="false"                     />
    <arg name="dev_names"       default="/dev/ttyACM0"              />
    <arg name="baud"            default="115200"                    />
    <arg name="address"         default="128"                       />
    <arg name="loop_hz"         default="10"                        />
    <arg name="deadman_secs"    default="3"                         />
    <arg name="speed_cmd_topic" default="$(arg name)/speed_command" />
    <arg name="stats_topic"     default="$(arg name)/stats"         />

    <node name="$(arg name)" pkg="roboclaw_driver" type="roboclaw_node.py">
        <param name="test_mode" type="bool" value="$(arg test_mode)" />
        <param name="dev_names" type="str" value="$(arg dev_names)" />
        <param name="baud" type="int" value="$(arg baud)" />
        <param name="address" type="int" value="$(arg address)" />
        <param name="loop_hz" type="int" value="$(arg loop_hz)" />
        <param name="deadman_secs" type="int" value="$(arg deadman_secs)" />
 
        <param name="speed_cmd_topic" type="str" value="$(arg speed_cmd_topic)" />
        <param name="stats_topic" type="str" value="$(arg stats_topic)" />
    </node>

</launch>